crates/swc_ecma_usage_analyzer/src/analyzer/mod.rs

```diff
@@ -1214,13 +1214,35 @@ where
                 } = e
                 {
                     // TODO: merge with may_have_side_effects
-                    let can_ignore =
-                        matches!(&**init, Expr::Call(call) if call.span.has_mark(marks.pure));
+
                     let id = id.to_id();
-                    self.used_recursively
-                        .insert(id.clone(), RecursiveUsage::Var { can_ignore });
-                    e.init.visit_with(&mut *self.with_ctx(ctx));
-                    self.used_recursively.remove(&id);
+
+                    match init {
```

Comment:
@Austaras Did you mean something like this?

Reply:
I can't see changes, seems like they're force pushed.